Tests/EndpointsTests/EndpointsTests.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  EndpointsTests.swift
+//  EndpointsTests
+//
+//  Created by Simon Jarbrant on 2019-10-29.
+//
+
+import XCTest
+import EndpointsTesting
+
+@testable import Endpoints
+
+struct TestMessage: Decodable {
+    let message: String
+}
+
+class EndpointsTests: XCTestCase {
+    func testEndpointCreatesValidURLRequest() throws {
+        let endpoint = TestEndpoint()
+        let request = try endpoint.asURLRequest()
+
+        XCTAssertEqual(request.url?.absoluteString, "https://example.com/message")
+        XCTAssertEqual(request.httpMethod, "GET")
+    }
+
+    func testCommunicatorCompletesSuccessfulRequest() throws {
+        let testTransporter = TestTransporter(responses: [
+            TestTransporter.TestResponse(code: 200, data: TestFixtures.simpleMessageData)
+        ])
+
+        let communicator = APICommunicator(transport: testTransporter)
+
+        let communicatorCompletionExpectation = XCTestExpectation(description: "Communicator completion expectation")
+        var expectedResult: Result<CommunicatorResponse<TestMessage>, Error>?
+        communicator.performRequest(to: TestEndpoint()) { result in
+            expectedResult = result
+            communicatorCompletionExpectation.fulfill()
+        }
+
+        wait(for: [communicatorCompletionExpectation], timeout: 1.0)
+
+        XCTAssertNotNil(expectedResult)
+        guard let unwrappedResult = expectedResult else {
```

Comment:
Kanske skulle man kunna använda denna för att slippa asserta och guard-letta:
`try XCTUnwrap(expectedResult, message: "Expected expectedResult to be non-optional")`